modules/phase_field/src/ics/SmoothCircleFromFileIC.C

```diff
@@ -0,0 +1,80 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "SmoothCircleFromFileIC.h"
+
+template <>
+InputParameters
+validParams<SmoothCircleFromFileIC>()
+{
+  InputParameters params = validParams<SmoothCircleBaseIC>();
+  params.addClassDescription("Multiple smooth circles read from a text file");
+  params.addRequiredParam<FileName>("file_name", "File containing circle centers and radii");
+
+  return params;
+}
+
+SmoothCircleFromFileIC::SmoothCircleFromFileIC(const InputParameters & parameters)
+  : SmoothCircleBaseIC(parameters),
+    _data(0),
+    _col_map(4),
+    _file_name(getParam<FileName>("file_name")),
+    _txt_reader(_file_name, true, " ", &_communicator),
+    _n_circles(0)
+{
+  // Read names and vectors from file
+  _txt_reader.read();
+  _col_names = _txt_reader.getColumnNames();
+  _data = _txt_reader.getColumnData();
+  _n_circles = _data[0].size();
+
+  // Check that the file has all the correct information
+  _col_map = {-1, -1, -1, -1};
```

Comment:
You can actually move this line right up to the constructor. That replaces the somewhat awkward resize to "4" you have.